### PR TITLE
Debug dispatch_to_batch

### PR DESCRIPTION
**Pull request by avrohomgottlieb**
## Issue Number

Related to Issue https://github.com/AlexsLemonade/scpca-portal/issues/944

## Purpose/Implementation Notes

After debugging the Batch implementation in PR https://github.com/AlexsLemonade/scpca-portal/pull/945, the final step in order to complete the Batch functionality was to debug the batch job submission workflow in the `dispatch_to_batch` management command. This PR fixed a handful of bugs in the implementation, tweaking the command itself as well as the underlying batch infrastructure.

## Types of changes

What types of changes does your code introduce?

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Functional tests

N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes

## Screenshots

N/A

### CODE REVIEW

**Comment by avrohomgottlieb** [comment]
One bug that was discovered is that the input library metadata file is meant to be included in the output computed file amount the other data files for spatial downloads. When the loading of metadata and generation of computed files happen on the same machine, as has been the case up until now, metadata files are loaded when objects are being created and data files are loaded during computed file generation. However, because the loading of metadata and the generation of computed files happen now on separate machines (the former on the API and the latter on fargate instances through batch), the metadata file wasn't being included with the other data files in the output computed file. I've addressed this bug in my most recent commit.

File: api/scpca_portal/management/commands/dispatch_to_batch.py [hunk 1]
```diff
@@ -28,19 +31,23 @@ def handle(self, *args, **kwargs):
...
         resource_id = project_id if project_id else sample_id
-        job_name = f"{resource_id} - {download_config_name}"
+        job_name = f"{resource_id}-{download_config_name}"
 
         response = batch.submit_job(
             jobName=job_name,
-            jobQueue="scpca_portal_project",
-            jobDefinition="scpca_portal_project",
+            jobQueue=os.environ.get("AWS_BATCH_JOB_QUEUE_NAME"),
+            jobDefinition=os.environ.get("AWS_BATCH_JOB_DEFINITION_NAME"),
```
Comment (davidsmejia) [inline api/scpca_portal/management/commands/dispatch_to_batch.py:50]: Let's move this to settings and import from there.

File: bin/sportal [hunk 2]
```diff
@@ -35,6 +35,9 @@ commands = {
     "create-portal-metadata": run_api.format("./manage.py create_portal_metadata {}"),
     "load-data": run_api.format("./manage.py load_data {}"),
     "mock-data": run_api.format("./manage.py load_mock_data {}"),
+    "load-metadata": run_api.format("./manage.py load_metadata {}"),
+    "generate-computed-files": run_api.format("./manage.py generate_computed_files {}"),
+    "dispatch-to-batch": run_api.format("./manage.py dispatch_to_batch {}"),
```
Comment (davidsmejia) [inline bin/sportal:40]:
```suggestion
```

We shouldn't try to dispatch to batch from local via sportal

File: api/scpca_portal/models/library.py [hunk 3]
```diff
@@ -91,13 +91,13 @@ def get_data_file_paths(cls, data: Dict, s3_input_bucket: str) -> List[Path]:
...
-        data_file_paths = [
-            file_path
-            for file_path in s3.list_input_paths(relative_path, s3_input_bucket)
-            if "metadata" not in file_path.name
-        ]
+        file_paths = s3.list_input_paths(relative_path, s3_input_bucket)
+
+        # input metadata json is included in spatial downloads, but excluded from all others
+        if Library.get_modality_from_file_paths(file_paths) == Library.Modalities.SPATIAL:
+            return file_paths
```
Comment (davidsmejia) [inline api/scpca_portal/models/library.py:98]:
```suggestion
        # input metadata json is excluded from single_cell downloads
        if Library.get_modality_from_file_paths(file_paths) == Library.Modalities.SINGLE_CELL:
            return [file_path for file_path in file_paths if "metadata" not in file_path.name]
```

Inverting the exception here would make it easier to maintain. This allows you to pair the logic for the exception and the return value in the same statement.

File: api/scpca_portal/models/library.py [hunk 4]
```diff
@@ -91,13 +91,13 @@ def get_data_file_paths(cls, data: Dict, s3_input_bucket: str) -> List[Path]:
...
 
-        return data_file_paths
+        return [file_path for file_path in file_paths if "metadata" not in file_path.name]
```
Comment (davidsmejia) [inline api/scpca_portal/models/library.py:100]:
```suggestion
        return file_paths
```

File: bin/sportal [hunk 5]
```diff
@@ -35,6 +35,9 @@ commands = {
     "create-portal-metadata": run_api.format("./manage.py create_portal_metadata {}"),
     "load-data": run_api.format("./manage.py load_data {}"),
     "mock-data": run_api.format("./manage.py load_mock_data {}"),
```
Comment (davidsmejia) [inline bin/sportal:37]: can you delete this while you are in here, this command doesn't exist anymore

File: api/scpca_portal/config/production.py [hunk 6]
```diff
@@ -16,12 +16,16 @@ class Production(Common):
     UPDATE_S3_DATA = True
 
     # AWS
-    AWS_REGION = os.getenv("AWS_REGION")
+    AWS_REGION = os.getenv("AWS_REGION", "us-east-1")
```
Comment (davidsmejia) [inline api/scpca_portal/config/production.py]: Let's move this to common.

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**